### ISSUE TITLE
tailsitter mc mode , Uncommanded alt_ref and thrust decreased,

### ISSUE THREAD

**Issue by MikeWellC** [PX4/Firmware#9032]
position mode , during flight, thrust decreased ,why?
https://logs.px4.io/plot_app?log=c90e6575-ce6b-4610-b182-86f6fb00b2dd

**Comment by RomanBapst** [comment]
@MikeWellC Hi! The problem you are facing is related to insufficient control authority in the pitch axis of the duorotor. As you can see in the logs the vehicle is having a hard time to get it's nose up.
This problem occurs if the vehicle is pitching up/down and therefore picking up ground speed. At some point the relative wind seen by the wing is a combination of wind due to movement and propeller slipstream. If the angle of attack seen by the wing is beyond the angle of attack for stalling the elevons become much less effective and they cannot produce the moment to bring the wing upright.

Now, the reason why the throttle cut is related to how the position controller works.
The position controller will try to turn it's thrust vector into the direction it want to accelerate. In your case it wanted to have the thrust vector pointing up but the actual thrust vector was till pointing forward and as a result it cut thrust.
In other words the position controller assumes that the vehicle can follow it's target attitude reasonably well, which is not the case for this vehicle.

The tragic part about this is that this vehicle can only pitch up if it has a considerable amount of throttle or if it has sufficient airspeed.

**Comment by jingego** [comment]
I am wondering could we set some min_thr for duo-rotors tail-sitter. such as for duo-rotors tail-sitter, if the pitch error larger than 30 degrees, the minimum thrust will be set to 0.6 to maintain the induced velocity. RC thrust may also be considered so that landing will not be affected. 
like 
<pre>if vehicle is duo-rotors tailsitter  
      min_thr=(pitch_error/0.5)*0.5*(rc.thrust/0.5); 
      constrain min_thr [0, 0.8]</pre>
if pitch_error is small (0.1) and the rc command (0.5) is to maintain current altitude, the min_thr=0.1;
if  pitch_error is large (0.5) and the rc command (0.5) is to maintain current altitude, the min_thr=0.5;
if  pitch_error is large (0.5) and the rc command (0.1) is to land, the min_thr=0.1;
if both are large(1&1), the min_thr=2, which will be force to 0.8;
if both are small(0.1&0.1), the min_thr=0.02;
and final use [min_thr, max_thr] to constrain thrust sp.

**Comment by VTOLDavid** [comment]
@RomanBapst @MikeWellC Hi! got the same problem yesterday flying my quad-rotor tailsitter in posittion mode. A suddently throttle out during a 25km/h wind landing at the same time that pitch was far from pitch sp. Luckily I managed to change to Fw mode and land in stabilize mode without problems. 
If it can be controled in stabilize mode by a human it shoud be done by the autopilot. 
I think that the position control in tailsitters should control thrust to control height and pitch and roll to control x and y axis that is the way it is done manually and works very well. 

The fact that the position controller cuts throttle when the thust vector is not the desired seems to me a serious bug. In a normal quadcopter it is not a problem because has great pitch autority but if the center of gravity is not well placed or has great speed and aerodynamic forces turn multicopter it probably will fall as our tailsitters do.

The solution that @jingego proposes should work but will be better if we avoid that behaviour from the position controller, it will be an improvement for all airframes.

**Comment by VTOLDavid** [comment]
This could be solved setting the MC_AIRMODE to 1?

**Comment by VTOLDavid** [comment]
Tested yesterday. Solves this problem, but if the aircraft does not have enough pitch authority with certain wind in position mode goes up forever...
